configuration/src/main/java/io/airlift/configuration/ConfigurationFactory.java

```diff
@@ -559,7 +559,14 @@ private static Object coerce(Class<?> type, String value)
                 return value;
             }
             if (Boolean.class == type || boolean.class == type) {
-                return Boolean.valueOf(value);
+                // Boolean.valueOf returns `false` when called with `"true "` argument
+                if ("true".equalsIgnoreCase(value)) {
+                    return Boolean.TRUE;
+                }
+                if ("false".equalsIgnoreCase(value)) {
+                    return Boolean.FALSE;
+                }
+                return null;
```

Comment:
We may want to fail in this case

Reply:
This will be turned into a failure later.